code/modules/surgery/plastic_surgery.dm

```diff
@@ -8,20 +8,29 @@
 	name = "reshape face"
 	implements = list(/obj/item/scalpel = 100, /obj/item/kitchen/knife = 50, TOOL_WIRECUTTER = 35)
 	time = 64
+	var/chosen_name
 
 /datum/surgery_step/reshape_face/preop(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
+	chosen_name = stripped_input(user, "Enter the name that [target] will have, post-surgery.", "Input a name", target.real_name, MAX_NAME_LEN)
+
+	if(QDELETED(tool) || QDELETED(target) || QDELETED(user) || !chosen_name)
+		return -1 // FAILURE
+
 	user.visible_message("[user] begins to alter [target]'s appearance.", "<span class='notice'>You begin to alter [target]'s appearance...</span>")
 
 /datum/surgery_step/reshape_face/success(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
-	if(target.has_trait(TRAIT_DISFIGURED, TRAIT_GENERIC))
-		target.remove_trait(TRAIT_DISFIGURED, TRAIT_GENERIC)
-		user.visible_message("[user] successfully restores [target]'s appearance!", "<span class='notice'>You successfully restore [target]'s appearance.</span>")
-	else
-		var/oldname = target.real_name
-		target.real_name = target.dna.species.random_name(target.gender,1)
-		var/newname = target.real_name	//something about how the code handles names required that I use this instead of target.real_name
-		user.visible_message("[user] alters [oldname]'s appearance completely, [target.p_they()] is now [newname]!", "<span class='notice'>You alter [oldname]'s appearance completely, [target.p_they()] is now [newname].</span>")
+	var/oldname = target.real_name
+	user.visible_message("[user] alters [oldname]'s appearance completely, [target.p_they()] is now [chosen_name]!", "<span class='notice'>You alter [oldname]'s appearance completely, [target.p_they()] is now [chosen_name].</span>")

```

Comment:
are

Reply:
What does p_they() actually do that just saying they doesn't?

Reply:
p_they() takes into account the mob's gender.

there has to be a p_are() to make the sentence grammatically correct no matter the mob's gender.

Reply:
just default to "they are now [name]"

Reply:
wait is p_they = he/she/it?